client/allocrunner/alloc_runner.go

```diff
@@ -547,40 +546,63 @@ func (ar *allocRunner) handleTaskStateUpdates() {
 			}
 		}
 
-		// if all live runners are sidecars - kill alloc
-		if killEvent == nil && hasSidecars && !hasNonSidecarTasks(liveRunners) {
-			killEvent = structs.NewTaskEvent(structs.TaskMainDead)
-		}
-
-		// If there's a kill event set and live runners, kill them
-		if killEvent != nil && len(liveRunners) > 0 {
-
-			// Log kill reason
-			switch killEvent.Type {
-			case structs.TaskLeaderDead:
-				ar.logger.Debug("leader task dead, destroying all tasks", "leader_task", killTask)
-			case structs.TaskMainDead:
-				ar.logger.Debug("main tasks dead, destroying all sidecar tasks")
-			default:
-				ar.logger.Debug("task failure, destroying all tasks", "failed_task", killTask)
+		if len(liveRunners) > 0 {
+			// if all live runners are sidecars - kill alloc
+			if killEvent == nil && hasSidecars && !hasNonSidecarTasks(liveRunners) {
```

Comment:
could the double negative here be wrapped in a `hasLiveSidecars(...)` helper?

Reply:
Hum...I don't think it can be extracted into a method because the live or not condition is determined by the method input: here it's passing a map of live runners, so the result will be live runners that are not sidecars.

I could move it to a intermediate variable is more descriptive: 

```go
hasLiveSidecars := hasSidecars && !hasNonSidecarTasks(liveRunners)
if killEvent == nil && hasLiveSidecars {
	killEvent = structs.NewTaskEvent(structs.TaskMainDead)
}
```

Does this look better?